rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,416 @@
+to restore the file associated with this bundle and fail if it cannot.
+
+### Get Deleted Bundle API
+
+The following table proposes modification of the existing path 
+[/bundle/{uuid}](https://github.com/HumanCellAtlas/data-store/blob/3d73935692f2030e7c19e4ec3b074a15361d33ca/dss-api.yml#L1189-L1449).
+to communicate the deletion of a bundle:
+
+**Path**: `GET /bundles/{uuid}`
+
+|Response Code| Description|
+|--------------|------------|
+|410| The bundle has been deleted. The reason and details for the deletion are returned. |
+
+**Path**: `PUT /bundles/{uuid}`
+
+|Response Code| Description|
+|--------------|------------|
+|409| A bundle or bundle tombstone with the same UUID and version already exists. |
+
+### Get Deleted File API
+
+The following proposes modification of the existing path 
+[/file/{uuid}](https://github.com/HumanCellAtlas/data-store/blob/3d73935692f2030e7c19e4ec3b074a15361d33ca/dss-api.yml#L237-L547)
+to communicate the deletion of a file:
+
+**Path**: `HEAD /file/{uuid}`
+
+|Response Code| Description|
+|--------------|------------|
+|410| The file has been deleted. The reason and details for the deletion are returned. |
+
+**Path**: `GET /file/{uuid}`
+
+|Response Code| Description|
+|--------------|------------|
+|410| The file has been deleted. The reason and details for the deletion are returned. |
+
+**Path**: `PUT /bundles/{uuid}`
+
+|Response Code| Description|
+|--------------|------------|
+|409| Returned when a file or file tombstone with the same UUID and/or version already exists.|
+
+
+## Deletion Process
+
+The following process describes the process for deleting data from the DSS. The same process is shared for both file and
+bundle deletes. It will be explicitly mentioned in areas where the processes diverge.
+
+Once it is determined that (meta)data needs to be deleted –
+the details of this determination are the scope of an SOP – the following
+procedures takes place: 
+
+1) A user identifies the Data Store file or bundle bundle to be deleted.
+
+2) The user makes a request to the appropriate API to retrieve the confirmation code. A list of bundles and/or files
+   affected is also returned in the request.
+
+3) The users makes a second request that included the confirmation code. This starts the deletion process.
+
+4) After the deletion request has been confirmed the DSS places **tombstones** in the
+   underlying storage system(S3 or GCP bucket). For a **logical deletion** request, tombstones are placed for all bundles
+   mentioned in the requests response. For **physical deletion** requests, tombstones are placed for all files and 
+   bundles mentioned in the request response. The tombstone's content is
+   as follows:
+   ```JSON
+   { "reason": "<reason>",
+     "details": "<additional info>",
+     "email": "<requester's email>"
+   }
+   ```
+   
+   A versioned bundle tombstone will cause `GET /bundle/{uuid}` to return a 410 for that bundle version. If the tombstoned
```

Comment:
Here and elsewhere, formally it should be "return a 410 status code"

Reply:
will fix